test/types.tsx

```diff
@@ -196,3 +196,55 @@ radio('option', 'a');
 radio({ name: 'option', value: 'a' });
 checkbox({ name: 'option' });
 checkbox({ name: 'option', value: 1 });
+
+// Raw Input
+
+const DatePicker: React.FC<{
+  onChange(value: Date): void;
+  value: string;
+}> = ({ onChange }) => (
+  <input
+    type="date"
+    onChange={e => onChange(new Date(JSON.stringify(e.target.value)))}
+  />
+);
+
+function RawInputTyped() {
+  const [formState, { raw }] = useFormState<{ name: string; date: string }>();
+  return (
+    <>
+      <input {...raw('name')} />
+      <DatePicker
+        {...raw({
```

Comment:
The raw value is now inferred from the types of `DatePicker` automatically - in this case it will be `Date`. It can also be set explicitly:

```ts
<DatePicker
  {...raw<Date>({
    // ...  
```